### DISCUSSION TITLE
trigger pi pico buzzer remotely

### DISCUSSION THREAD

**Discussion by siennamediatest** [micropython/micropython#10678] [RP2040 / Pico]
I work in a retirement home and have built a 'dementia proof' media centre for residents. I've also built an IR remote using pi pico w, which only has 2 buttons and it much simpler for them to use. The remote has an onboard buzzer, and I want to be able to trigger the buzzer from a gpio button on the media centre to make the remote easier to locate. I have done a few web server tests to remotely turn on an led, but I'm having difficulty modifying the approach to work with the buzzer. Ideally I would want to remotely run a micropython code that I wrote so that the buzzer plays music.

**Comment by peterhinch** [comment]
As a first step just play a tone using PWM. You're already using `uasyncio` so have the code that flashes the LED issue `create_task()` as below.
```python
from machine import Pin, PWM
pwmpin = Pin(16, Pin.OUT)
pwm = PWM(pwmpin)
async def beep():  # Use asyncio.create_task(beep()) to run this
    pwm.init(freq=440, duty_u16=32768)
    await asyncio.sleep(1)
    pwm.deinit()
```

**Reply by willlahr** [reply]
does it need to be uasyncio on a pico rather han asyncio?

**Reply by robert-hh** [reply]
uasyncio is an alias to asyncio, or the opposite. You can use either one.

**Reply by peterhinch** [reply]
Yes. My code is a follow-up to [this post](https://github.com/peterhinch/micropython_ir/issues/23#issuecomment-1416064713) so I knew that the OP is using this alias.

In general I use this code style:
```python
import uasyncio as asyncio
```
because it facilitates writing code which is portable between MicroPython and CPython:
```python
try:
    import uasyncio as asyncio
except ImportError:
    import asyncio  # Running on CPython
```

**Reply by siennamediatest** [reply]
I figured it out! I used a basic tutorial to set up a web server to control an led. Then I changed the code so instead of the led it would activate the buzzer, and then removed the website part so the buzzer beeps once the client connects to the IP address. Now I just need to change it so the buzzer plays music instead of a loud beep.

**Reply by peterhinch** [reply]
You need to define what you mean by "music". The simplest option is a sequence of tones at different frequencies. Synthesising polyphonic music (chords) is tougher. If you want true audio rendition of stored files (e.g. .WAV) you need to investigate I2S. For MP3 you'll need extra hardware [VS1053b](https://github.com/peterhinch/micropython-vs1053).

**Reply by siennamediatest** [reply]
I created a simple library of tones so the buzzer plays 'ob la di'!
